docs: Core Color class (#41)
NativeScript/docs: content/guide/core/color.md

---
title: Color
description: Create a color object with any color representation and use it in your NativeScript apps.
---

> This class allows you to create a color object with all color components(alpha, red, green, blue in the [0, 255] range) from different types of color representations such as, ARGB, color name, hex, etc. The properties `ios` and `android` of an instance of this class provide the color object for the respective native platform.

Reviewer: This class enables the creation of a color object with color components (alpha, red, green, blue) ranging from 0 to 255, using various color representations such as ARGB, color names, hex values, and more. The properties ios and android of an instance of this class provide the corresponding color object for the respective native platforms.

> This class allows you to create a color object with all color components(alpha, red, green, blue in the [0, 255] range) from different types of color representations such as, ARGB, color name, hex, etc. The properties `ios` and `android` of an instance of this class provide the color object for the respective native platform.

Reviewer 2: ```suggestion
This `Color` class enables the creation of a color object with color components (alpha, red, green, blue) ranging from 0 to 255, using various color representations such as ARGB, color names, hex values, and more. The properties `ios` and `android` of an instance of this class provide the corresponding color object for the respective native platforms such as [UIColor](https://developer.apple.com/documentation/uikit/uicolor) on iOS and [Color ints](https://developer.android.com/reference/android/graphics/Color) on Android.
```

> This `Color` class enables the creation of a color object with color components (alpha, red, green, blue) ranging from 0 to 255, using various color representations such as ARGB, color names, hex values, and more. The properties `ios` and `android` of an instance of this class provide the corresponding color object for the respective native platforms such as [UIColor](https://developer.apple.com/documentation/uikit/uicolor) on iOS and [Color ints](https://developer.android.com/reference/android/graphics/Color) on Android.

Reviewer 3: ```suggestion
The `Color` class enables the creation of a color object:
 - using color components (alpha, red, green, blue) ranging from 0 to 255,
 - using various color representations like ARGB, color names, hex values, and more.
 
The `ios` and `android` properties of the `Color` class instance return the native [UIColor](https://developer.apple.com/documentation/uikit/uicolor) on iOS and [Color](https://developer.android.com/reference/android/graphics/Color) on Android.
```

Note: `using various color representations like ARGB, color names, hex values, and more.` we should define `and more` or remove it.

Suggestion:
 - split each option as a new line/list item and link them within the page. ie -> color names should link to a section about using color names etc.

```ts
const color = new Color(alpha: number, red: number, green:number, blue: number, type?: 'rgb' | 'hsl' | 'hsv')
```
---

### a

```ts
colorAlpha: number = color.a
```

Gets the Alpha component of the color. This is a `read-only` property.

---

### r

```ts
colorRed: number = color.r
```

Gets the Red component of the color. This is a `read-only` property.

---

### g

```ts
colorGreen: number = color.g
```

Gets the Green component of the color. This is a `read-only` property.

---

### b

```ts
colorBlue: number = color.b
```

Gets the Blue component of the color. This is a `read-only` property.

---

### argb

```ts
colorARGB : number = color.argb
```

Gets the Argb Number representation of this color where each 8 bits represent a single color component. This is a `read-only` property.

---

### hex

```ts
colorHex: string = color.hex
```

Gets the Hexadecimal string representation of the color.

---

### name

```ts
colorName: string =color.name
```

Gets the known name of this instance. Defined only if it has been constructed from a known color name - e.g. "red".

---

### android

```ts
androidColor = color.android
```

> Gets the android-specific integer value representation. Same as the Argb one.

Reviewer: `ARGB` one

> Gets the android-specific integer value representation. Same as the Argb one.

Reviewer 2: ```suggestion
Gets the android-specific integer value representation. Same as the `ARGB` one.
```
